File: core/src/bara/game/components/ChildrenComponent.java
package bara.game.components;

import com.badlogic.ashley.core.Component;
import com.badlogic.ashley.core.Entity;
import com.badlogic.gdx.utils.Pool;

import java.util.HashMap;

public class ChildrenComponent implements Component, Pool.Poolable {
    //NOTE 8/13/17: use enums/ints for keys instead?
    public HashMap<String, Entity> children = new HashMap<String, Entity>();

    @Override
    public void reset() {
        children.clear();
    }

}

File: core/src/bara/game/components/ChildrenComponent.kt
package bara.game.components

import bara.game.ashley.GameComponent
import com.badlogic.ashley.core.Entity
import com.badlogic.gdx.utils.ObjectMap
import ktx.ashley.EngineEntity


class ChildrenComponent : GameComponent() {
    //NOTE 8/13/17: use enums/ints for keys instead?
    var children = ObjectMap<String, Entity>()

    override fun reset() {
        children.clear()
    }

    override fun buildOnto(engineEntity: EngineEntity) {

    }

}
